Moda.Integrations/src/Moda.Integrations.MicrosoftGraph/MicrosoftGraphService.cs

```diff
@@ -63,19 +64,37 @@ public async Task<Result<IEnumerable<IExternalEmployee>>> GetEmployees(Cancellat
     /// <returns></returns>
     private async Task<List<User>> GetGroupMembers(string groupId, CancellationToken cancellationToken)
     {
-        // TODO handle paging
         var members = await _graphServiceClient.Groups[groupId]
             .TransitiveMembers
             .GetAsync(requestConfiguration =>
             {
                 requestConfiguration.Headers.Add("ConsistencyLevel", "eventual");
-                requestConfiguration.QueryParameters.Count = true;
                 requestConfiguration.QueryParameters.Expand = new string[] { "manager($select=id)" };
                 requestConfiguration.QueryParameters.Select = _selectOptions;
                 requestConfiguration.QueryParameters.Filter = "accountEnabled eq true and usertype eq 'Member'";
+                requestConfiguration.QueryParameters.Top = _maxPageSize;
             }, cancellationToken);
 
-        return members?.OdataCount > 0 ? members.Value!.Select(m => (User)m).ToList() : new List<User>();
+        List<User> users = new();
+        if (members is null || members.Value is null)
+        {
+            _logger.LogWarning("GetGroupMembers:  No members found for group {GroupId} in Active Directory via Microsoft Graph", groupId);
+            return users;
+        }
+
+        var pageIterator = PageIterator<DirectoryObject, DirectoryObjectCollectionResponse>
+            .CreatePageIterator(
+                _graphServiceClient,
+                members,
+                (d) =>
+                {
+                    users.Add((User)d);
+                    return true;
+                });
```

Comment:
so, not sure if it's needed, but the `requestConfiguration.Headers.Add` from the initial call will not persist if the iterator decides it needs to make another request, there's another delegate you can pass after this one to readd that.

Reply:
Adding that header enabled advanced query capabilities like the not equals (ne), but since that isn't working after we added the expand it should be okay to remove it.

https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=csharp